### ISSUE TITLE
[FEATURE] watch folders

### ISSUE THREAD

**Issue by creativetags** [ImageOptim/ImageOptim#162]
It would be sweet to be able to drag a folder from the Finder and have it continually watched for new images which would then be processed in the background

**Comment by pornel** [comment]
Yup. That's a good idea. Do you have a suggestion for discoverable and uncomplicated UI for it?

Currently dropping a folder optimizes content of the folder once.

**Comment by creativetags** [comment]
I think have another window for "Watch Folders" that would list folders to watch. You could then drag folders to be watched onto that window. I'd suggest only showing this window if it was opened from a menu item rather than when the app opens to keep things simple for newbies.

**Comment by blairanderson** [comment]
Bumping this! 

i'm often taking screenshots, finding things to take screenshots of and sending those screenshots around. 

I've changed my screenshot folder specifically being its a constantly in flux. 

within imageoptim i'd love to select that folder and have all photos immediately processed. 

Anyone in the community created scripts to make this happen?

**Comment by ThermScissorpunch** [comment]
I would love this :-)

**Comment by woefi** [comment]
The workflow with other automatic processing tools is usually: 
`10 watch <source-folder>`
`20 process all files inside `
`30 write them to the <target-folder>`
`40 delete originals`
`50 goto 10`

But ImageOptim doesn't work this way, as it processes files in place.
So how to keep track which files were already optimized?

**Comment by NiklasBr** [comment]
> So how to keep track which files were already optimized?

We don't need to track them. [Mac OS has events ](https://dougscripts.com/itunes/itinfo/folderaction01.php)*on file added to folder* we can listen to to only get new images to process.

Example

```applescript
on adding folder items to theFolder after receiving addedItems
    -- Specify the target application
    set targetApp to "ImageOptim"
    
    -- Loop through added items
    repeat with anItem in addedItems
        set itemPath to POSIX path of anItem
        if itemPath ends with ".jpg" or itemPath ends with ".png" or itemPath ends with ".gif" or itemPath ends with ".jpeg" then
            tell application targetApp
                open anItem
            end tell
        end if
    end repeat
end adding folder items to
```
